client/grpc_client.go

```diff
@@ -99,7 +122,7 @@ func (cli *grpcClient) Error() error {
 
 func (cli *grpcClient) EchoAsync(msg string) *ReqRes {
 	req := types.ToRequestEcho(msg)
-	res, err := cli.client.Echo(context.Background(), req.GetEcho())
+	res, err := cli.client.Echo(context.Background(), req.GetEcho(), grpc.FailFast(true))
 	if err != nil {
 		cli.err = err
```

Comment:
Wouldn't it be better to just call StopForError() rather than checkErrGetResult() & checkGetErr()?